### PR TITLE
Updated hybrid key format.

### PR DESCRIPTION
**Pull request by xvzcf**

### CODE REVIEW

File: ssl/statem/ext_oqs_extra.h [hunk 1]
```diff
@@ -1,62 +1,74 @@
...
+                                     uint16_t* classical_msg_len,
+                                     unsigned char** pq_msg,
+                                     uint16_t* pq_msg_len) {
+
+  int ec_curve_id = OQS_KEM_CLASSICAL_CURVEID(group_id);
+  unsigned int pq_kem_id = OQS_KEM_NID(group_id);
+
+  /* Generate a dummy key to get the size of an encoded EC public key
+   * TODO(oqs): There must be a better way than this */
+  EVP_PKEY* key = ssl_generate_pkey_group(NULL, ec_curve_id);
```
Comment (dstebila) [inline ssl/statem/ext_oqs_extra.h]: Do you need to free this before returning?

File: ssl/statem/ext_oqs_extra.h [hunk 2]
```diff
@@ -1,62 +1,74 @@
...
+  if (key == NULL) {
+      return 0;
+  }
+  unsigned char *throwaway;
```
Comment (dstebila) [inline ssl/statem/ext_oqs_extra.h]: Do you need to free this before returning?

**Review by dstebila — CHANGES_REQUESTED** [review]

File: ssl/statem/ext_oqs_extra.h (hunk 2 above)
Comment (xvzcf) [reply ssl/statem/ext_oqs_extra.h]: Yes to both, I just put it up in the hopes someone would suggest a way without any of these allocations.

Comment (dstebila) [reply ssl/statem/ext_oqs_extra.h]: Not that I can think of, but I didn't check the EVP API to see if there's a function specifically for this purpose.  (You can probably get this information directly from crypto/ec, but that may pierce an abstraction layer that you don't want to...)

Comment (xvzcf) [reply ssl/statem/ext_oqs_extra.h]: @baentsch any ideas?

Comment (baentsch) [reply ssl/statem/ext_oqs_extra.h]: Sorry, not immediately: Getting a PKEY and then querying parameters (incl pub key len) is the only pattern I know using EVP ([also perpetuated in OSSL3](https://www.openssl.org/docs/manmaster/man3/EVP_PKEY_get1_encoded_public_key.html) as far as I can tell). Only by using EC APIs directly you may do other things (didn't look into that, though as I still don't like the "concept" OQS keys == EC keys :-) 

Second thought: Isn't there a classic PKEY object "around" (to use/query) anyway when doing hybrid operations? Shouldn't the SSL connection object contain a PKEY reference/or require building one during session setup) anyway? Why is there a need to create _this_ (new) throwaway object in the first place? A cursory glance at the `ssl/statem/extensions_clnt.c` lets me think there should be PKEY objects to query (and if a hybrid PKEY, why shouldn't that contain classic size information to query)?

File: ssl/statem/ext_oqs_extra.h [hunk 3]
```diff
@@ -1,62 +1,76 @@
...
+  *classical_msg_len = EVP_PKEY_get1_tls_encodedpoint(key, &throwaway);
```
Comment (baentsch) [inline ssl/statem/ext_oqs_extra.h]: The more I think about it, why not better create and use here a new `OQS_KEM_CLASSICAL_KEYLEN(<oqs-hybrid-group-id>) macro? If I'm not wrong, it only has three different alternative values to return...

Comment (xvzcf) [reply ssl/statem/ext_oqs_extra.h]: I ended up just doing this; we can make it a macro if we start finding that we need it in multiple places.

**Review by dstebila — APPROVED** [review]

**Pull request merged**